### PR TITLE
Handle deprecations from Symfony 5.1

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tickets       | N/A
| License       | MIT
| Doc PR        | N/A

This PR addresses method signatures of the Symfony Config and DependencyInjection components that have changed in Symfony 5.1.

### CODE REVIEW

File: src/Bridge/Symfony/Bundle/Resources/config/api.xml [hunk 1]
```diff
@@ -18,7 +18,6 @@
         <service id="api_platform.operation_method_resolver" class="ApiPlatform\Core\Bridge\Symfony\Routing\OperationMethodResolver" public="false">
             <argument type="service" id="api_platform.router" />
             <argument type="service" id="api_platform.metadata.resource.metadata_factory" />
-            <deprecated>The "%service_id%" service is deprecated since API Platform 2.5.</deprecated>
```
Comment (derrabus) [inline src/Bridge/Symfony/Bundle/Resources/config/api.xml:21]: 💡 I haven't found a way to produce an XML file that can be loaded with DependencyInjection 4.4 and 5.1 without errors:
* DI 5.1 triggers a deprecation if the attributes `package` and `version` are missing.
* DI 4.4 throws an exception, if those attributes are present.

This is why I'm deprecating the services inside the extension now.

**Comment by alanpoulain** [comment]
Thank you very much @derrabus!

**Comment by filisko** [comment]
Hi everyone, I still get the error with api-platform/core v2.5.6. Do you know why could it be?
```
  5x: Since symfony/dependency-injection 5.1: Not setting the attribute "version" of the node "deprecated" in "/project/vendor/api-platform/core/src/Bridge/Symfony/Bundle/DependencyInjection/../Resources/config/api.xml" is deprecated.

  5x: Since symfony/dependency-injection 5.1: Not setting the attribute "package" of the node "deprecated" in "/project/vendor/api-platform/core/src/Bridge/Symfony/Bundle/DependencyInjection/../Resources/config/api.xml" is deprecated.

  4x: Since symfony/config 5.1: The signature of method "Symfony\Component\Config\Definition\Builder\NodeDefinition::setDeprecated()" requires 3 arguments: "string $package, string $version, string $message", not defining them is deprecated.

  1x: Since symfony/dependency-injection 5.1: Not setting the attribute "package" of the node "deprecated" in "/project/vendor/api-platform/core/src/Bridge/Symfony/Bundle/DependencyInjection/../Resources/config/filter.xml" is deprecated.

  1x: Since symfony/dependency-injection 5.1: Not setting the attribute "version" of the node "deprecated" in "/project/vendor/api-platform/core/src/Bridge/Symfony/Bundle/DependencyInjection/../Resources/config/filter.xml" is deprecated.
```

**Comment by stloyd** [comment]
Cause this was not yet released: https://github.com/api-platform/core/compare/v2.5.6...2.5

**Comment by filisko** [comment]
Oh sorry, i don't know why i thought that because it is merged is released... Thank you @stloyd !!
Do you know when it would it be released? :thinking:

**Comment by soyuka** [comment]
Summer and holidays are making things to move at slow pace, bear with us :)

**Comment by alebedev80** [comment]
When do you plan to release it?

**Pull request merged**